src/components/games/GameControlsMod.svelte

```diff
@@ -70,7 +75,31 @@
     });
   }
 
+  function getNewestVersion(versions: string[]): string {
+    const versionsCopy = [...versions];
+
+    return (
+      versionsCopy
+        .sort((a, b) => {
+          const aParts = a.split(".").map(Number);
+          const bParts = b.split(".").map(Number);
+
+          for (let i = 0; i < Math.max(aParts.length, bParts.length); i++) {
+            const aPart = aParts[i] || 0;
+            const bPart = bParts[i] || 0;
+
+            if (aPart !== bPart) {
+              return aPart - bPart;
+            }
+          }
+          return 0;
+        })
+        .pop() || ""
+    );
+  }
```

Comment:
this shouldn't be required to be this involved.  the version list off the source is already ordered from latest to oldest,  So you should be able to just grab the first item in `modVersionList`.

Also if you did have to parse semver, you should use the already included `semver` library instead of rolling it yourself.

Probably still a good idea to wrap it in a function so you can return something like `""` is the list is empty but you should be able to simplify this substantially.